### PR TITLE
RONDB-338: Golang Profiling/Refactoring Integration tests

### PR DESCRIPTION
**Pull request by olapiv**
Recommend to look at commit by commit

* Large refactoring in integration tests package
* Added ping benchmarking test
* Letting each go-routine (re-)use it's own gRPC connection for benchmarking tests
* Setting GOMAXPROCS for integration tests
* More granular latency calculations

Did some other experiments on the way (which were reverted), but they essentially lead to the refactoring commit.

### CODE REVIEW

File: storage/ndb/rest-server/rest-api-server/internal/integrationtests/batchpkread/wrapper_test.go [hunk 1]
```diff
@@ -52,5 +53,13 @@ func TestMain(m *testing.M) {
 	}
 	defer cleanup()
 
+	// Benchmarking pkreads tends to perform better with more client-side go-routines
```
Comment (smkniazi) [inline storage/ndb/rest-server/rest-api-server/internal/integrationtests/batchpkread/wrapper_test.go:56]: For testing runtime.GOMAXPROCS is also called in integrationtests.InitialiseTesting. 

Move this code before you call InitializeTesting and set the conf.Internal.GOMAXPROC

File: storage/ndb/rest-server/rest-api-server/internal/integrationtests/ping/wrapper_test.go [hunk 2]
```diff
@@ -52,5 +53,8 @@ func TestMain(m *testing.M) {
 	}
 	defer cleanup()
 
+	// Less than NumCPUs seemed to work well here:
+	runtime.GOMAXPROCS(conf.Internal.GOMAXPROCS)
```
Comment (smkniazi) [inline storage/ndb/rest-server/rest-api-server/internal/integrationtests/ping/wrapper_test.go]: set the conf.Internal.GOMAXPROCS before calling integrationtests.InitialiseTesting

File: storage/ndb/rest-server/rest-api-server/internal/integrationtests/pkread/utils.go [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package pkread
+
+import (
+	"testing"
+
+	"hopsworks.ai/rdrs/pkg/api"
+)
+
+func pkTestMultiple(t *testing.T, tests map[string]api.PKTestInfo, isBinaryData bool) {
```
Comment (smkniazi) [inline storage/ndb/rest-server/rest-api-server/internal/integrationtests/pkread/utils.go:9]: Multiple? Does pkTestAll sound better?

File: storage/ndb/rest-server/rest-api-server/internal/integrationtests/batchpkread/performance_test.go [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+	"google.golang.org/grpc"
+	"hopsworks.ai/rdrs/internal/config"
+	"hopsworks.ai/rdrs/internal/integrationtests/testclient"
+	"hopsworks.ai/rdrs/internal/testutils"
+	"hopsworks.ai/rdrs/pkg/api"
+	"hopsworks.ai/rdrs/resources/testdbs"
+)
+
+/*
+	The number of parallel client go-routines spawned can be influenced by setting
```
Comment (smkniazi) [inline storage/ndb/rest-server/rest-api-server/internal/integrationtests/batchpkread/performance_test.go]: Bit confusing, GOMAXPROCS only controls the number of runtime threads. These threads can handle any number of go routines.

File: storage/ndb/rest-server/rest-api-server/internal/integrationtests/ping/performance_test.go [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+)
+
+/*
+	The number of parallel client go-routines spawned can be influenced by setting
+	runtime.GOMAXPROCS(). It defaults to the number of CPUs.
+
+	go test \
+		-test.bench BenchmarkSimple \
+		-test.run=thisexpressionwontmatchanytest \
+		-cpu 1,2,4,8 \
```
Comment (smkniazi) [inline storage/ndb/rest-server/rest-api-server/internal/integrationtests/ping/performance_test.go:37]: does -cpu 1,2,4,8 has any impact? 
sometimes you override this with
runtime.GOMAXPROCS(runtime.NumCPU() * 5)

**Review by smkniazi — APPROVED** [review]
Looks OK to me. Approved with some minor comments.

File: storage/ndb/rest-server/rest-api-server/internal/integrationtests/ping/wrapper_test.go (hunk 2 above)
Comment (olapiv) [reply storage/ndb/rest-server/rest-api-server/internal/integrationtests/ping/wrapper_test.go]: Yeah it's actually entirely unnecessary here.

File: storage/ndb/rest-server/rest-api-server/internal/integrationtests/ping/performance_test.go (hunk 5 above)
Comment (olapiv) [reply storage/ndb/rest-server/rest-api-server/internal/integrationtests/ping/performance_test.go:37]: Haven't used it so far!

File: storage/ndb/rest-server/rest-api-server/internal/integrationtests/batchpkread/wrapper_test.go (hunk 1 above)
Comment (olapiv) [reply storage/ndb/rest-server/rest-api-server/internal/integrationtests/batchpkread/wrapper_test.go:56]: Well it's supposed to overwrite the standard initialisation here.

**Pull request merged**